### ISSUE TITLE
Can't open subdirectory in new window

### ISSUE THREAD

**Issue by stevensdavid** [zed-industries/zed#22184]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When an editor is open in directory `X`, Zed refuses to open any subdirectory `Y` of `X` and instead just moves focus to the editor opened to `X`. This happens both when opening Zed from a terminal and when using an empty Zed window's "Open folder" modal, and makes it more difficult to work with subprojects.

### Zed Version and System Specs

Zed: v0.165.4 (Zed)
OS: macOS 15.1.1
Memory: 32 GiB
Architecture: aarch64

**Comment by osiewicz** [comment]
FWIW it does work as you'd expect when a subdirectory is opened from an existing project (so not from an empty Zed window).

**Comment by stevensdavid** [comment]
I don't think that this seems to be reproducible anymore, at least not when following my own instructions. Thanks!
